sw/device/tests/entropy_src_fw_override_test.c

```diff
@@ -258,13 +275,83 @@ static status_t entropy_config(
   return OK_STATUS();
 }
 
+// Configure the entropy complex.
+static status_t reenable_entropy_src(
+    dif_entropy_src_single_bit_mode_t single_bit_mode,
+    bool bypass_conditioner) {
+  uint32_t data;
+  bool entropy_req_intr = false;
+  size_t words_written = 0;
+  uint32_t kInputMsg[kEntropyFifoBufferSize];
+  uint32_t cnt = 0;
+
+  // Don't let the extract and insert interrupt handler run until we've
+  // re-enabled the entropy source.
+  irq_external_ctrl(false);
+
+  // Make ENTROPY_SRC produce new seeds until CSRNG has no outstanding
+  // entropy requests.
+  do {
+    // If the FW_OV_WR_DATA register is not ready yet,
+    // we wait for a bit and try again.
+    data = mmio_region_read32(entropy_src.base_addr,
+                              ENTROPY_SRC_FW_OV_WR_FIFO_FULL_REG_OFFSET);
+    if (data) {
+      busy_spin_micros(10);
+      continue;
+    }
+    // Make SHA3 absorb new data and produce a new seed right away.
+    // We use a counter for the input message to avoid triggering
+    // ES_BUS_CMP_ALERT.
+    CHECK_DIF_OK(dif_entropy_src_fw_override_sha3_start_insert(
+        &entropy_src, kDifToggleEnabled));
+    for (int i = 0; i < kEntropyFifoBufferSize; i++) {
+      kInputMsg[i] = cnt++;
+    }
+    CHECK_DIF_OK(dif_entropy_src_fw_ov_data_write(
+        &entropy_src, (const uint32_t *)kInputMsg, kEntropyFifoBufferSize,
+        &words_written));
+    CHECK_DIF_OK(dif_entropy_src_fw_override_sha3_start_insert(
+        &entropy_src, kDifToggleDisabled));
```

Comment:
Nit: this triggers the SHA3 operation which may take up to 48 (CS AES Halt interface) + 25 (padding) + 24 (SHA3 rounds) cycles to finish. If the CS_ENTROPY_REQ bit is still high after executing this DIF function it's most likely because the SHA3 hasn't finished yet. To avoid this, you could poll the FSM state register of the ENTROPY_SRC before acking and checking the interrupt again.

But it's not that dramatic, worst case you just run this loop here once more and "produce" one seed that is not necessary.